util/genrest/resttools/error_response.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resttools
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"google.golang.org/api/googleapi"
+	"google.golang.org/grpc/codes"
+)
+
+// Derived from internal source: go/http-canonical-mapping.
+var GRPCToHTTP map[codes.Code]int = map[codes.Code]int{
```

Comment:
A nit on principle, unlikely to come up in practice:

Would it be better to have the table be an internal variable, and have a `func GRPCToHttp(code codes.Code) int` so that if there isn't a key with the value of `code`, we treat is as `codes.Unknown`?

Reply:
Yeah great idea! I'll add that.